### ISSUE TITLE
Latest stable and unstable sees sporadic reboots with watchdog enabled

### ISSUE THREAD

**Issue by peterhoeg** [NixOS/nixpkgs#131920]
### Describe the bug

For the longest time we've been running both physical and virtual instances with the systemd watchdog enabled

```
  systemd.extraConfig = ''
    RuntimeWatchdogSec=1m
    ShutdownWatchdogSec=1m
  '';
```

*However*, since b361dcf0bd3610d1fea6b44d72404add188da51d was merged and made it into both unstable as well as 21.05 through a subsequent backport, *something* goes wrong with systemd that makes it not send the watchdog keep-alives and the machines as a consequence get force rebooted after a period of time.

Without the mentioned commit on both unstable and 21.05, the configuration above works as expected.

*With* the commit and the watchdog enabled, machines will reboot anywhere between after 2 and 8 minutes after boot.

### Steps To Reproduce
1. Run latest 21.05 or unstable
2. Activate the `RuntimeWatchdog`
3. Reboot
4. Wait a few minutes for the machine to force reboot

We *really* should pin this issue.

### Notify maintainers
@dasJ @eelco @andir @flokli @kloenk

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - systemd
# a list of nixos modules affected by the problem
module:
 - systemd
```

**Comment by peterhoeg** [comment]
This was observed on 3 physical devices as well as 2 VMs.

**Comment by flokli** [comment]
I don't see how just applying this CVE patch could cause this regression, and if it were, how it's a NixOS-specific issue.

Is there anything suspicious in the logs? Are other distros running into this too, and maybe there's an upstream issue for it?

**Comment by peterhoeg** [comment]
I haven't been able to find anything elsewhere about this problem and re-enabling the watchdog consistently reproduces the problem. NixOS's systemd is customized - so maybe our customizations together with this change is what triggers it.

I would love to hear from others who are running with the watchdog. 

I'll set up a proper test environment to try this out. Need to get to the bottom of it.

**Comment by chivay** [comment]
I'm not sure if it's the same bug, but a watchdog-related regression was recently reported to Linux kernel.
See: https://bugzilla.kernel.org/show_bug.cgi?id=213809

**Comment by flokli** [comment]
@peterhoeg can you see if applying that attached patch to your kernel (via `boot.kernelPatches`) fixes this for you, and if so, see if it landed/will land in linux-stable?

**Comment by collares** [comment]
The offending patch was reverted in linux-stable: https://lore.kernel.org/linux-watchdog/YQzWl9GeD6OgxzWA@kroah.com/. If this is the cause, it should be fixed in 5.4.139, 5.10.57 or 5.13.9 (but note that the 5.12 series is EOL and won't get the fix).

**Comment by Atemu** [comment]
The PRs have been merged and are in unstable now, is the issue fixed @peterhoeg?
